src/hotspot/share/opto/vectornode.cpp

```diff
@@ -1851,36 +1851,28 @@ Node* NegVNode::Ideal(PhaseGVN* phase, bool can_reshape) {
   return NULL;
 }
 
-Node* ReverseBytesVNode::Identity(PhaseGVN* phase) {
-  if (is_predicated_using_blend()) {
-    return this;
-  }
-  // ReverseBytesV (ReverseBytesV X , MASK) , MASK =>  X
-  if (in(1)->Opcode() == Op_ReverseBytesV) {
-    if (is_predicated_vector() && in(1)->is_predicated_vector() && in(2) == in(1)->in(2)) {
-      return in(1)->in(1);
-    } else {
-      // ReverseBytesV (ReverseBytesV X) =>  X
-      return in(1)->in(1);
+static Node* reverse_operations_identity(Node* n, Node* in1) {
+  if (n->is_predicated_using_blend()) {
+    return n;
+  }
```

Comment:
The change in this patch looks fine to me! Just a concern the previous codes, for patterns like:
```
VectorBlend X (ReverseBytesV (ReverseBytesV Y)) MASK
```
we will miss the transformation to:
```
VectorBlend X Y MASK
```
right? Thanks!

Reply:
No, since the flag Predicated_with_blend is set over nodes if operation itself is predicated, here both ReverseBytesV nodes are non-predicated ones and this check is part of identity routines of reverse* operations.

Reply:
Oh, right! What I mean is the case like:
```
VectorBlend (ReverseBytesV X) (ReverseBytesV (ReverseBytesV X)) MASK   ==> VectorBlend (ReverseBytesV X) X MASK 
```
which is the same case with `test_reversebytes_long_transform2` for non-predicated systems.

Reply:
As already [discussed](https://github.com/openjdk/panama-vector/pull/182#discussion_r927419799),  we can handle this as a separate PR along with other complimentary operations. This bug fix patch is fixing a specific issue.

Reply:
Agree, thanks!